src/openssl/digest.h

```diff
@@ -25,7 +25,14 @@
 
 struct digest {
 	const EVP_MD *digest;
+#if OPENSSL_VERSION_MAJOR < 3
 	HMAC_CTX *hmac_ctx;
+#else
+	EVP_MAC_CTX *hmac_ctx;
+	void *hmac_key;
+	size_t hmac_key_len;
+	const char *hmac_algo;
```

Comment:
These things can be stored inside `hmac_ctx`, so no need to keep copies (I hope)?